File: siruseri.py
up_pitch = []
low_pitch = []

n = int(input('Enter no. of singers:'))



for i in range(n):
    print(f'Enter the lower limit of singer {i+1}: ')
    l = int(input())
    low_pitch.append(l)
    print(f'Enter the higher limit of singer {i+1}:')
    h = int(input())
    up_pitch.append(h)

pitch = list(zip(low_pitch,up_pitch))


def win_check(ls,n):
    k = 0
    score = [0] * n
    for i in ls:
        for j in ls:
            if i[0] < j[0]:           #checking lower limit of pitch
                score[k] += 2
            elif i[0] == j[0]:        #if lower limit same then check higher limit.
                if i[1] > j[1]:       #Assuming draw isn't possible as per the question.
                    score[k] += 2
        k += 1
    print('final score:')
    print(score)

win_check(pitch,n)

File: siruseri.c
#include<stdio.h>
int score_check(int[],int[],int[],int);

int main()
{
  int low[10],high[10],n,score[10];
  printf("Enter the no of singers:");
  scanf("%d\n", &n );
  for(int i = 0; i < n; i++)
  {
    printf("Enter the lower limit of singer %d: ", i+1);
    scanf("%d\n", &low[i] );

    printf("Enter the higher limit of singer %d: ", i+1);
    scanf("%d\n", &high[i] );
  }

  score_check(low, high, score, n);
  for(int i = 0; i < n; i++)
  {
    printf(score[i]);
  }
  return 0;
}

int score_check(int a[],int b[],int c[],int n)
{
  for(int i = 0; i < n ; i++)
  {
    for(int j = 0; j < n; j++)
    {
      if(a[i] < a[j])
      {
        c[i] += 2;
      }
      else if(a[i] == a[j])
      {
        if(b[i] > b[j])
        {
          c[i] += 2;
        }
      }
    }
    return 0;
  }

}
